### ISSUE TITLE
Add continuous integration on hardware

### ISSUE THREAD

**Issue by wingo** [Igalia/snabb#259]
Currently our continuous integration tests are run by TravisCI and run on generic software data paths.  We need to set up continuous integration on dedicated hardware.  This will also allow us to do performance testing; see https://github.com/Igalia/snabbswitch/issues/110.  Probably we can integrate this with SnabbBot.

**Comment by kbara** [comment]
CI would be ideal. A script that runs a bunch of tests that's either conditionally run from 'make test' whenever hardware is detected, or in the worst case run by hand, would still be a non-trivial step forward.

**Comment by dpino** [comment]
This task depends on https://github.com/snabblab/snabblab-nixos/issues/13

**Comment by dpino** [comment]
Actually #13 was not a blocker.  Now I have managed to run `scripts/snabb_bot.sh` is our machines. Snabb-bot needs a TEST_IMAGE to run. The instructions to build the test environment are here: https://github.com/eugeneia/snabbswitch-docker. I got stuck while running `make image` https://github.com/eugeneia/snabbswitch-docker/issues/1

**Comment by dpino** [comment]
I made some progress related with this.

I configured SnabbBot so it pulls from igalia/snabbswitch and merges each PR in the PR tray against the branch "lwaftr". The name of the bot is  "snabb1-lwaftr/snabb-test". For instance:

https://github.com/Igalia/snabbswitch/pull/289

Don't pay to other bots such as "snabb1-eugeneia/snabb-nfv-test" or "snabb1-dpino/snabb-nfv-test", these were testing bots.

With regard to the CI test we were running before, the same test is run by SnabbBot's continuous integration test (make test_ci). This target runs all files with a selftest and all scripts names selftest.sh, such as program/lwaftr/tests/end-to-end/selftest.sh.

Point is at this moment all PR's are failing (can see here as well https://gist.github.com/dpino, SnabbBot logs are reported as gists of the user that run the SnabbBot). Basically they cannot merge with current "lwaftr" branch. I think this is the right branch where to integrate, as it's our development branch now. Once this problem is solved we can think of what other tests we would like our SnabbBot to run.
